File: src/main/java/net/Chidoziealways/everythingjapanese/screen/custom/pedestal/PedestalScreen.java
package net.Chidoziealways.everythingjapanese.screen.custom.pedestal;

import com.mojang.blaze3d.systems.RenderSystem;
import net.Chidoziealways.everythingjapanese.EverythingJapanese;
import net.minecraft.client.gui.GuiGraphics;
import net.minecraft.client.gui.screens.inventory.AbstractContainerScreen;
import net.minecraft.client.renderer.GameRenderer;
import net.minecraft.client.renderer.RenderType;
import net.minecraft.network.chat.Component;
import net.minecraft.resources.ResourceLocation;
import net.minecraft.world.entity.player.Inventory;

public class PedestalScreen extends AbstractContainerScreen<PedestalMenu> {
    private static final ResourceLocation GUI_TEXTURE =
            ResourceLocation.fromNamespaceAndPath(EverythingJapanese.MOD_ID, "textures/gui/pedestal/pedestal_gui.png");

    public PedestalScreen(PedestalMenu pMenu, Inventory pPlayerInventory, Component pTitle) {
        super(pMenu, pPlayerInventory, pTitle);
    }

    @Override
    protected void renderBg(GuiGraphics gui, float partialTicks, int mouseX, int mouseY) {
        // 1) Compute top‑left corner to center the GUI
        int x = (this.width  - this.imageWidth ) / 2;
        int y = (this.height - this.imageHeight) / 2;

        // 2) Draw the full background in one call
        //    – no more setShader/setShaderTexture calls
        //    – blit now takes a RenderType supplier + texture + coords + texture‑size
        gui.blit(
                RenderType::guiTextured,      // use the built‑in GUI shader/type :contentReference[oaicite:0]{index=0}
                GUI_TEXTURE,                  // your GUI texture ResourceLocation
                x, y,                         // screen position
                0, 0,                         // u/v start in the texture
                imageWidth, imageHeight,      // width/height to draw
                256, 256                      // full PNG size of your texture :contentReference[oaicite:1]{index=1}
        );
    }

    @Override
    public void render(GuiGraphics pGuiGraphics, int pMouseX, int pMouseY, float pPartialTick) {
        super.render(pGuiGraphics, pMouseX, pMouseY, pPartialTick);
        renderTooltip(pGuiGraphics, pMouseX, pMouseY);
    }
}

File: src/main/kotlin/net/Chidoziealways/everythingjapanese/screen/custom/pedestal/PedestalScreen.kt
package net.Chidoziealways.everythingjapanese.screen.custom.pedestal

import net.Chidoziealways.everythingjapanese.JAPANESE_MOD_ID
import net.minecraft.client.gui.GuiGraphicsExtractor
import net.minecraft.client.gui.screens.inventory.AbstractContainerScreen
import net.minecraft.client.renderer.RenderPipelines
import net.minecraft.network.chat.Component
import net.minecraft.resources.Identifier
import net.minecraft.world.entity.player.Inventory

class PedestalScreen(pMenu: PedestalMenu, pPlayerInventory: Inventory, pTitle: Component) :
    AbstractContainerScreen<PedestalMenu>(pMenu, pPlayerInventory, pTitle) {

        init {
            //imageWidth = 256
            //imageHeight = 256
        }

    override fun extractBackground(gui: GuiGraphicsExtractor, mouseX: Int, mouseY: Int, partialTicks: Float) {
        // 1) Compute top‑left corner to center the GUI
        val x = (this.width - this.imageWidth) / 2
        val y = (this.height - this.imageHeight) / 2

        // 2) Draw the full background in one call
        //    – no more setShader/setShaderTexture calls
        //    – blit now takes a RenderType supplier + texture + coords + texture‑size
        gui.blit(
            RenderPipelines.GUI_TEXTURED,
            GUI_TEXTURE,  // your GUI texture Identifier
            x, y,  // screen position
            0F, 0F,  // u/v start in the texture
            imageWidth, imageHeight,  // width/height to draw
            256, 256 // full PNG size of your texture :contentReference[oaicite:1]{index=1}
        )
    }

    override fun extractRenderState(pGuiGraphics: GuiGraphicsExtractor, pMouseX: Int, pMouseY: Int, pPartialTick: Float) {
        super.extractRenderState(pGuiGraphics, pMouseX, pMouseY, pPartialTick)
        extractTooltip(pGuiGraphics, pMouseX, pMouseY)
    }

    companion object {
        private val GUI_TEXTURE: Identifier =
            Identifier.fromNamespaceAndPath(JAPANESE_MOD_ID, "textures/gui/pedestal/pedestal_gui.png")
    }
}
